pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -389,3 +408,10 @@ func (c *Controller) deletePodAutoscaler(ns, name string) error {
 	}
 	return nil
 }
+
+func getDeploymentName() string {
```

Comment:
Tooling already exists for that: https://github.com/DataDog/datadog-agent/blob/main/pkg/util/kubernetes/apiserver/common/common_linux.go

It reads `DD_POD_NAME` which is already set and defaults to `os.hostname()` (should work for DCA). From there you can use a variant of https://github.com/DataDog/datadog-agent/blob/main/pkg/util/kubernetes/helpers.go#L22-L40 to get Deployment name from POD Name

Reply:
Thanks! Made a new helper to get the deployment name from the pod name which reuses the code from the helper you shared here